service/manager_test.go

```diff
@@ -307,6 +308,43 @@ func TestStartManager_WithErrors(t *testing.T) {
 	assert.Error(t, control.ServiceError)
 }
 
+func TestStartManager_WithMultipleErrors(t *testing.T) {
+	s := makeManager()
+
+	moduleProvider1 := &StubModuleProvider{
+		NameVal: "stubModule1",
+		CreateVal: func(host Host) (Module, error) {
+			return &StubModule{
+				Host:       host,
+				StartError: errors.New("can't start stubModule1"),
+			}, nil
+		},
+	}
+	require.NoError(t, s.addModule(moduleProvider1))
+
+	moduleProvider2 := &StubModuleProvider{
+		NameVal: "stubModule2",
+		CreateVal: func(host Host) (Module, error) {
+			return &StubModule{
+				Host:       host,
+				StartError: errors.New("can't start stubModule2"),
+			}, nil
+		},
+	}
+	require.NoError(t, s.addModule(moduleProvider2))
+
+	time.AfterFunc(10*time.Second, func() {
+		log.Fatalf("Service dint shut down on its own for over 10 secs so forcefully killing it!")
```

Comment:
Interesting, it never auto timed out for me. Guessing because the default timeout is 10 mins(https://golang.org/cmd/go/#hdr-Description_of_testing_flags) and I usually gave up waiting in ~ 5mins.  Anyway updated this to t.Fatalf, good suggestion.

Reply:
Commented before thorough testing my bad. Anyway using t.Fatalf doesn't kill the underlying stuck go process running manager.go so have to use log.Fatalf. My understanding is t.Fatalf calls FailNow which only kills the running test but does not stop other goroutines created during the test. Lmk if you know a better way to handle this.